fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobManager.java

```diff
@@ -0,0 +1,278 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.load;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.io.Writable;
+import com.starrocks.persist.CreateInsertOverwriteJobInfo;
+import com.starrocks.persist.InsertOverwriteStateChangeInfo;
+import com.starrocks.persist.gson.GsonPostProcessable;
+import com.starrocks.persist.gson.GsonUtils;
+import com.starrocks.server.GlobalStateMgr;
+import io.netty.util.concurrent.DefaultThreadFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class InsertOverwriteJobManager implements Writable, GsonPostProcessable {
```

Comment:
if we have 1000 insert into overwrite jobs in FE, how many FE memories will use at most for the job metadata?

Reply:
In my test, one InsertOverwriteJob will take up 56 bytes. and 1000 jobs will take up almost 56kbyes.

Reply:
@kangkaisen PTAL